specs/ssh_configuration.adoc

```diff
@@ -0,0 +1,282 @@
+is the process of setup a password less ssh between the provisioning node and other
+cluster nodes.
+
+SSH utilities are helps tendrl in SSH public-key configuration process. Each SSH
+utility have some different functionality.
+
+1. Utility for generate SSH-Key for particular user.
+2. Utility for copy authorized ssh key.
+3. Utility for find sshd service port number and status.
+
+Later entire workflow is tied together with flows, which are not part of this
+specification.
+
+== Use Cases
+
+Needs SSH configuration utility in common to support SSH configuration for
+provisioning cluster.
+
+== Proposed change
+
+
+SSH public-key configuration have four major step, each step is a separate utility
+method that always works locally on a specific node. Each of these utility methods
+are executed by ansible in local node. Later entire workflow is tied together with
+flows, which are not part of this specification.
+
+1. Generate a key-pair for a user. The method can take the user name and
+   key-pair name as optional parameters, whose defaults are root and tendrl
+   respectively.
+2. Install an SSH key into a user's authorized_keys. SSH public key should be
+   a mandatory parameter, while the user name should be optional. User name
+   should default to root.
+3. Check if the SSH daemon is running and get it's port number.
+4. Check if the SSH incoming port is open. (It will implemented in future not part
+   of this spec).
+
+== Alternatives
+
+None
+
+== Data model impact
+
+None
+
+== Impacted modules
+
+=== Tendrl/common impact
+
+* Create a new ansible module in commons utils to create a new user with
+  SSH-key. If the user already present then it should create only SSH-key for that
+  user.
+```
+Example arguments for the ansible user module is:
+Create a 2048-bit SSH key for user tendrl in /home/tendrl/.ssh/id_rsa.pub
+- user:
+    name: tendrl (default root)
+    generate_ssh_key: yes
+    ssh_key_bits: 2048
+    ssh_key_file: .ssh/id_rsa.pub
+```
+* Create a new ansible module commons utils to add SSH authorized key
+```
+Example arguments for the ansible authorized_key module is:
+- name: Set authorized key took from file
+  authorized_key:
+    user: tendrl (default root)
+    key: "pass pair of ssh-keys"
+    (default path is "(homedir)+/.ssh/authorized_keys")
+    path: "Alternate path to the authorized_keys file" (optional)
+```
+* Create a new a function in utils to find ssh daemon status and port number.
+  (In implementation section it is explained briefly)
+
+=== Tendrl/node_agent impact
+None
+
+=== Tendrl/gluster_integration impact
+None
+
+=== Tendrl/ceph_integration impact
+None
+
+=== Tendrl/alerting impact
+None
+
+=== Tendrl/performance_monitoring impact
+None
+
+== Security impact
+None
+
+== Notifications/Monitoring impact
+None
+
+== Other end user impact
+None
+
+== Performance Impact
+None
+
+== Other deployer impact
+None
+
+== Developer impact
+None
```

Comment:
Can you please add how a developer will be using these utils, By adding some code snippets